src/dotnet/commands/dotnet-build/Program.cs

```diff
@@ -33,7 +33,7 @@ public static int Run(string[] args)
             }
         }
 
-        private static bool OnExecute(List<ProjectContext> contexts, CompilerCommandApp args)
+        private static bool OnExecute(List<ProjectContext> contexts, NativeCompilerCommandApp args)
```

Comment:
Indeed, that looks so horrible. I'll just clone the code in the --no-dependency change